Signal/src/ViewControllers/ContactShareViewHelper.swift

```diff
@@ -0,0 +1,289 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+import SignalServiceKit
+import ContactsUI
+import MessageUI
+
+@objc
+public protocol ContactShareViewHelperDelegate: class {
+    func didCreateOrEditContact()
+}
+
+@objc
+public class ContactShareViewHelper: NSObject, CNContactViewControllerDelegate {
```

Comment:
I pulled out a helper class that contains to the logic for interacting with contact shares.